### PR TITLE
[main] Adding triggers to azure-pipeline-pr

### PR DESCRIPTION
**Pull request by AlitzelMendez**
After some conversations we run public post-merge builds for this pipeline, so readding trigger so this actually happens

I will rename the pipeline to reflect better what it is doing

### CODE REVIEW

File: azure-pipelines-pr.yml [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+    include:
+    - main
+    - release/6.0
+    - release/7.0
+    - release/8.0
+  paths:
+    include:
+    - '*'
+    exclude:
+    - .github/*
```
Comment (dougbu) [inline azure-pipelines-pr.yml:13]: nit: the `*` is implied

File: azure-pipelines-pr.yml [hunk 2]
```diff
@@ -1,4 +1,26 @@
...
+    - Documentation/*
+    - Images/*
```
Comment (dougbu) [inline azure-pipelines-pr.yml:15]: nit: this folder doesn't exist

File: azure-pipelines-pr.yml [hunk 3]
```diff
@@ -1,4 +1,26 @@
...
+    - CODE-OF-CONDUCT.md
+    - CONTRIBUTING.md
+    - LICENSE.TXT
+    - PATENTS.TXT
```
Comment (dougbu) [inline azure-pipelines-pr.yml:19]: nit: this file doesn't exist

**Review by dougbu — APPROVED** [review]
I _think_ we'll need to backport this fix to the servicing branches

**Comment by AlitzelMendez** [comment]
> I _think_ we'll need to backport this fix to the servicing branches

I think every branch has different triggers/restrictions, so probably is better do this for every specific branch instead of backporting

**Review by dougbu — APPROVED** [review]

**Pull request merged**